src/enemy/gibdo.c

```diff
@@ -0,0 +1,434 @@
+    if (sub_080378B0(this) == 0) {
+        if (sub_08037914(this) == 0) {
+            this->field_0x74.HWORD--;
+            if (this->field_0x74.HWORD == 0) {
+                sub_08037794(this);
+            } else {
+                UpdateAnimationSingleFrame(this);
+                UpdateAnimationSingleFrame(this);
+                if (ProcessMovement(this) == 0) {
+                    this->field_0xf--;
+                    if (this->field_0xf == 0) {
+                        sub_080379BC(this);
+                    }
+                } else {
+                    this->actionDelay--;
+                    if (this->actionDelay == 0) {
+                        sub_0803797C(this);
+                    }
+                }
+            }
+        }
+    }
+}
+
+void sub_08037690(Entity* this) {
+    UpdateAnimationSingleFrame(this);
+    if ((this->frames.all & 0x80) != 0) {
+        this->field_0x76.HALF.HI = 0x14;
+        sub_08037794(this);
+    } else {
+        if ((this->frames.all & 1) != 0) {
+            this->damageType = 0x27;
+            ProcessMovement(this);
+        }
+    }
+}
+// Non-matching
+NONMATCH("asm/non_matching/gibdo/sub_080376D0.inc", void sub_080376D0(Entity* this)) {
+    u8* x;
+    if (sub_080379EC(this) == 0) {
+        ResetPlayer();
+        gPlayerState.field_0x1a[0] = gPlayerState.field_0x1a[0] | 0x80;
+        gPlayerState.field_0xa = gPlayerState.field_0xa | 0x80;
+        CopyPositionAndSpriteOffset(&gPlayerEntity, this);
+        UpdateAnimationSingleFrame(this);
+        x = &this->frames.all;
+        if ((*x & 0x1) != 0) {
+            this->field_0x7c.BYTES.byte0--;
+            if (this->field_0x7c.BYTES.byte0 == 0) {
+                sub_08037A58(this);
+            }
+        } else {
+            *x = *x ^ 1;
+            sub_08037B10(this);
+        }
+    }
+}
+END_NONMATCH
+
+void sub_0803773C(Entity* this) {
+    UpdateAnimationSingleFrame(this);
+    if ((this->frames.all & 0x80) != 0) {
+        sub_08037794(this);
+    }
+}
+// Turn into Stalfos
+void sub_0803775C(Entity* this) {
+    Entity* x;
+    this->actionDelay--;
+    if (this->actionDelay == 0) {
+        x = CreateEnemy(0x42, 0);
+        if (x != 0) {
+            sub_0804A4E4(this, x);
+            sub_08037C0C(this, x);
+        }
+        DeleteEntity(this);
+    }
+}
+
+void sub_08037794(Entity* this) {
+    this->action = 1;
+    this->field_0x74.HWORD = 0x1e;
+    this->damageType = 0x26;
+    InitAnimationForceUpdate(this, this->animationState);
+}
+
+void sub_080377B0(Entity* this) {
+    u32 r1;
+    u32 r2;
+    this->action = 2;
+    this->field_0xf = 8;
+    r1 = Random();
+    this->field_0x74.HWORD = (r1 & 0x38) + 0x78;
+    this->nonPlanarMovement = 0x40;
+    r2 = Random();
+    if ((sub_08049FA0(this) == 0) && ((r2 & 3) != 0)) {
+        this->direction = (sub_08049EE4(this) + 4) & 0x18;
+    } else {
+        this->direction = r2 & 0x18;
+    }
+    this->animationState = this->direction / 8;
```

Comment:
Many of these can be turned into proper macros.
see https://github.com/zeldaret/tmc/blob/master/include/entity.h#L202